akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/FutureDirectivesSpec.scala

```diff
@@ -80,18 +80,25 @@ class FutureDirectivesSpec extends RoutingSpec with Inside {
     }
     "fail fast if the circuit breaker is open" in new TestWithCircuitBreaker {
       openBreaker()
-      Get() ~> onCompleteWithBreaker(breaker)(Future.successful(1)) { echoComplete } ~> check {
-        inside(rejection) {
-          case CircuitBreakerOpenRejection(_) ⇒
-        }
-      }
+      // since this is timing sensitive, try a few times to observe the breaker open
+      awaitAssert(
+        Get() ~> onCompleteWithBreaker(breaker)(Future.successful(1)) { echoComplete } ~> check {
+          inside(rejection) {
+            case CircuitBreakerOpenRejection(_) ⇒
+          }
+        }, 500.millis)
```

Comment:
I think that can be even longer. Everything that is still significant shorter than the `breakerResetTimeout` would do. Maybe half the `breakerResetTimeout`?

Reply:
Good idea